app/controllers/posts_controller.rb

```diff
@@ -8,6 +8,41 @@ def index
   end
 
   def show
+    @user = current_user
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+    if @post.save
+      puts @post
+      redirect_to user_posts_path(current_user)
+    else
+      render :new
+    end
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    Like.new(author: current_user, post: @post).save
+    redirect_to user_posts_path(current_user)
+  end
```

Comment:
### **Great** implementation of the project requirements.

- [ ] Kindly create a ``LikesController`` action for post likes and refactor your code.

- A user should be able to add a like to posts. The likes controller should have a create actions

- There should be a route in the config/routes.rb file for creating like.